rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -328,7 +335,7 @@ impl<TPayload: Clone + Send + Sync + 'static> StreamProcessor<TPayload> {
 
                 if let Some(msg) = invalid_message {
                     tracing::error!(?message, "Invalid message");
-                    self.dlq_policy.produce(msg);
+                    self.consumer_state.lock().unwrap().dlq_policy.produce(msg);
```

Comment:
isn’t this in the same scope as the lock on line 226? which would mean that this can deadlock?

Reply:
Fixed, thanks!